horreum-backend/src/main/resources/db/changeLog.xml

```diff
@@ -2277,5 +2277,56 @@
                 USING (exists (SELECT 1 FROM test WHERE test.id = test_id AND has_role2(test.owner, 'tester')));
         </sql>
     </changeSet>
+
+    <changeSet id="55" author="jwhiting">
+        <createProcedure>
+            CREATE OR REPLACE FUNCTION after_run_update_func() RETURNS TRIGGER AS $$
+            DECLARE
+                v_schema text;
+                v_schemaid integer;
+                v_has_schema boolean := false;
+            BEGIN
+                FOR v_schema IN (SELECT jsonb_path_query(NEW.data, '$.\$schema'::jsonpath)#>>'{}') LOOP
+                    v_schemaid := (SELECT id FROM schema WHERE uri = v_schema);
+                    IF v_schemaid IS NOT NULL THEN
+                        INSERT INTO run_schemas (runid, testid, prefix, uri, schemaid)
+                            VALUES (NEW.id, NEW.testid, '$', v_schema, v_schemaid);
+                        v_has_schema := true;
+                    END IF;
+                END LOOP;
+                FOR v_schema IN (SELECT jsonb_path_query(NEW.data, '$.*.\$schema'::jsonpath)#>>'{}') LOOP
+                    v_schemaid := (SELECT id FROM schema WHERE uri = v_schema);
+                    IF v_schemaid IS NOT NULL THEN
+                        INSERT INTO run_schemas (runid, testid, prefix, uri, schemaid)
+                            VALUES (NEW.id, NEW.testid, '$.*', v_schema, v_schemaid);
+                        v_has_schema := true;
+                    END IF;
+                END LOOP;
+                IF NOT v_has_schema THEN
+                    PERFORM pg_notify('calculate_tags', auth_suffix(NEW.id::text) );
+                END IF;
+                DELETE FROM dataset WHERE runid = OLD.id;
+                PERFORM pg_notify('calculate_datasets', auth_suffix(NEW.id::text) );
+                RETURN NULL;
+            END;
+            $$ LANGUAGE plpgsql;
+        </createProcedure>
+        <createProcedure>
```

Comment:
I think that you should delete all existing datasets for given run in `before_run_update_func`, too.

Reply:
yes